src/dependencyChecks.js

```diff
@@ -0,0 +1,96 @@
+    let v1 = parseInt(matches[1])
+    let v2 = parseInt(matches[2])
+    if (v1>=3) {
+      log(`looks like your python is Python 3, unfortunately emsdk expects Python 2`)
+      return false
+    }
+  } else {
+    log('failed to detect python version. make sure python points to python 2')
+  }
+  return true
+}
+
+const getChecksWithCommands = (command) => [
+      ['cmake --version', 'cmake', `cmake 3.4.3 or newer not found. Try installing with '${command} cmake' and rerunning?`, cmakeVersionCheck],
+      ['python --version 2>&1', 'python', `python not found. Try installing with '${command} python' and rerunning?`, pythonVersionCheck],
+      ['curl --version', 'curl', `curl not found. Try installing with '${command} curl' and rerunning?`],
+      ['git --version', 'git', `git not found. Try installing with '${command} git' and rerunning?`],
+  ]
+
+function getChecksForDistro(os, distro) {
+  const distroChecks = {
+    darwin: [
+      [
+        "brew --version",
+        "brew",
+        "brew not found. Try installing at https://brew.sh and rerunning?"
+      ],
+      [
+        "rustup target add wasm32-unknown-emscripten",
+        "rustup",
+        "rustup not found. Try installing at https://rustup.rs and rerunning?"
+      ],
+      [
+        "cargo --version",
+        "cargo",
+        "cargo not found. Try installing at https://rustup.rs and rerunning?"
+      ]
+    ].concat(getChecksWithCommands("brew install")),
+    fedora: [
+      ['rustup target add wasm32-unknown-emscripten', 'rustup', 'rustup not found. Try installing at https://rustup.rs and rerunning?'],
+      ['cargo --version', 'cargo', 'cargo not found. Try installing at https://rustup.rs and rerunning?'],
+    ].concat(getChecksWithCommands('sudo dnf install')),
+    ubuntu: [
+      ['rustup target add wasm32-unknown-emscripten', 'rustup', 'rustup not found. Try installing at https://rustup.rs and rerunning?'],
+      ['cargo --version', 'cargo', 'cargo not found. Try installing at https://rustup.rs and rerunning?'],
+    ].concat(getChecksWithCommands('sudo apt-get')),
+    default: [
+      ['rustup target add wasm32-unknown-emscripten', 'rustup', 'rustup not found. Try installing at https://rustup.rs and rerunning?'],
+      ['cargo --version', 'cargo', 'cargo not found. Try installing at https://rustup.rs and rerunning?'],
+      ['cmake --version', 'cmake', 'cmake 3.4.3 or newer not found. Try installing cmake via your distributions package manager', cmakeVersionCheck],
+      ['python --version 2>&1', 'python', 'python not found. Try installing python via your distributions package manager', pythonVersionCheck],
+      ['curl --version', 'curl', 'curl not found. Try installing curl via your distributions package manager'],
+      ['git --version', 'git', 'git not found. Try installing git via your distributions package manager'],
+    ]
+  };
+  const lowerKeyOs = os.toLowerCase();
+  if (lowerKeyOs === "darwin") {
+    return distroChecks.darwin;
```

Comment:
Agreed that a separate file is good to prevent `setup.js` from getting too long! I'm not convinced we need all this complexity with lowerKeyOs though — is there any reason we can't just have `getChecksForDistro` contain a simple if...else if...else if...else chain?

Reply:
Or even a reason why we shouldn't just access and check `process.detailedos` in this function directly?

Reply:
Would also love to reduce duplication of commands in this code! Things like `rustup target add wasm32-unknown-emscripten` are repeated in a lot of places.

Reply:
I added the lowerCase stuff because there are a lot of distros out there. If somebody wants to add Arch Linux he/she can just add an attribute to the Object containing the configs.

Not accessing `process.detailedos` here decouples this function from that library. In case something else should be implemented.

I can add some `const string = 'rustup target add wasm32-unknown-emscripten'` stuff and then use string interpolation. But I will be quite busy this week.

Reply:
Oh, and thanks for pointing me to standard